crypto/s2n_pkey.c

```diff
@@ -68,19 +68,21 @@ int s2n_pkey_size(const struct s2n_pkey *pkey)
     return pkey->size(pkey);
 }
 
-int s2n_pkey_sign(const struct s2n_pkey *pkey, struct s2n_hash_state *digest, struct s2n_blob *signature)
+int s2n_pkey_sign(const struct s2n_pkey *pkey, s2n_signature_algorithm sig_alg,
```

Comment:
are there usages of s2n_pkey->sign() outside of s2n, and would changing this signature break them?

Reply:
If it's not in s2n.h, it's not a public API :)

Reply:
That was my first thought, but it kind of is. The s2n_pkey struct is in s2n.h, so technically customers could call s2n_pkey->sign(). I think this is a good point.

Reply:
Colm just pointed out offline that only the declaration is in the header, so the structure is opaque. Customers don't know anything about the implementation, so we can change whatever we want.